hedera-node/hedera-file-service-impl/src/main/java/com/hedera/node/app/service/file/impl/codec/FileServiceStateTranslator.java

```diff
@@ -0,0 +1,111 @@
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hedera.node.app.service.file.impl.codec;
+
+import static java.util.Objects.requireNonNull;
+
+import com.hedera.hapi.node.base.FileID;
+import com.hedera.hapi.node.base.Key;
+import com.hedera.hapi.node.state.file.File;
+import com.hedera.node.app.service.file.impl.ReadableFileStoreImpl;
+import com.hedera.node.app.service.mono.files.HFileMeta;
+import com.hedera.node.app.service.mono.pbj.PbjConverter;
+import com.hedera.pbj.runtime.io.buffer.Bytes;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import org.apache.commons.codec.DecoderException;
+
+/**
+ * The class is used to convert a {@link com.hedera.node.app.service.mono.files.HFileMeta} content and metadata  to a {@link File} and vice versa during the migration process
+ */
+public class FileServiceStateTranslator {
+
+    /**
+     * The method converts a {@link com.hedera.node.app.service.mono.files.HFileMeta} content and metadata  to a {@link File}
+     * @param fileID old protobuf fileID that will be used to search in old state
+     * @param hederaFs old state
+     * @return new protobuf File
+     */
+    @NonNull
+    public static File stateToPbj(
+            @NonNull com.hederahashgraph.api.proto.java.FileID fileID,
+            @NonNull com.hedera.node.app.service.mono.files.HederaFs hederaFs) {
+        requireNonNull(fileID);
+        return stateToPbj(hederaFs.cat(fileID), hederaFs.getattr(fileID), fileID);
+    }
+
+    /**
+     * The method converts a {@link com.hedera.node.app.service.mono.files.HFileMeta} content and metadata  to a {@link File}
+     * @param data content of the file in old state
+     * @param metadata metadata of the file in old state
+     * @param fileID old protobuf fileID that will be used in order to initialize the new protobuf file
+     * @return new protobuf File
+     */
+    @NonNull
+    public static File stateToPbj(
+            @NonNull final byte[] data,
+            @NonNull final com.hedera.node.app.service.mono.files.HFileMeta metadata,
+            @NonNull final com.hederahashgraph.api.proto.java.FileID fileID) {
+        requireNonNull(data);
+        requireNonNull(metadata);
+        requireNonNull(fileID);
+        final var fileBuilder = new File.Builder();
+        fileBuilder.fileNumber(fileID.getFileNum());
+        fileBuilder.expirationTime(metadata.getExpiry());
+        fileBuilder.keys(PbjConverter.asPbjKey(metadata.getWacl()).keyList());
```

Comment:
The ACL isn't a required item, so this should check for null from metadata.getWacl() and  not set the keys if the original object has no keys.

Reply:
fixing it